### ISSUE TITLE
Clarification Needed: Minimum Node.js Version for CI GitHub Action

### ISSUE THREAD

**Issue by carpasse** [expressjs/discussions#229]
**Main Issue**
We need to clarify the minimum version of Node.js that our repositories' Continuous Integration (CI) GitHub Actions are expected to run on.

**Context**
While working on the [initiative to add scorecards](https://github.com/expressjs/security-wg/issues/2) to Express.js organizations' repositories, we've been addressing any failing Continuous Integration (CI) GitHub Actions. However, we've encountered inconsistencies regarding the minimum version of Node.js required to run the CI tests.

For example:
- [mime-db](https://github.com/jshttp/mime-db/blob/master/.github/workflows/ci.yml#L13) sets Node.js 0.6 as its minimum version.
- [express](https://github.com/expressjs/express/blob/master/.github/workflows/ci.yml#L29) sets Node.js 0.10 as its minimum version.
- [final handler](https://github.com/pillarjs/finalhandler/blob/master/.github/workflows/ci.yml#L13) sets Node.js 0.8 as its minimum version.

We need to standardize the minimum Node.js version across all repositories to ensure consistency and compatibility.

**Proposal**
I propose opening this issue to discuss and decide on a unified minimum version of Node.js for all repositories' CI GitHub Actions within the Express.js organizations.

**Comment by wesleytodd** [comment]
All of those need to remain as is until we start updating them for major bumps. When we go to v5 for `express` most of those will likely land a few breaking changes as well as drop many node versions in the test matrix. 

@sheplu can you link to the current discsussion for the LTS strategy doc pr? I am not sure if it has moved from [this one](https://github.com/expressjs/discussions/pull/199) to the admin repo yet.

**Comment by carpasse** [comment]
@wesleytodd some repositories are still using [travis](https://github.com/carpasse/etag) Is it ok to migrate them to github actions respecting the specified Node.js versions?

**Comment by UlisesGascon** [comment]
> All of those need to remain as is until we start updating them for major bumps. When we go to v5 for `express` most of those will likely land a few breaking changes as well as drop many node versions in the test matrix. 


I think that for the minimum version, there's a different history, and we can establish a general rule before v5 is released, as it is an independent event.

- As a general rule, any dependency or sub-dependency of Express v4 should at least support Node 0.10, as it is the minimum supported version for Express@4 today.
- Any other project in the organization (including Express dependencies and sub-dependencies) that supports lower versions than 0.10 currently in their CI pipelines can serve as a reminder, just in case other projects or users depend on it. So in this context final handler supports Node@0.8 and mime-db supports Node@06
- Once Express@5 is released, the minimum Node.js version will obviously change for the next major version of those libraries that the project depends on.

Side note: Dropping support for any specific Node.js version is considered a major semver change, while extending support to new ones is a minor semver change (AFAIK).

> some repositories are still using travis Is it ok to migrate them to github actions respecting the specified Node.js versions?

Migrating to GitHub Actions is the way.

**Comment by wesleytodd** [comment]
> I think that for the minimum version, there's a different history, and we can establish a general rule before v5 is released

I agree we can set a rule, and as long as we keep to your point made in the "side note" (many of those packages are used outside of express and we cannot just change the supported node versions without a major bump) then I am good.

> Is it ok to migrate them to github actions respecting the specified Node.js versions?
 > Migrating to GitHub Actions is the way.

Yep absolutely!!!
